pkg/apis/alicloud/validation/infrastructure.go

```diff
@@ -97,8 +98,48 @@ func ValidateInfrastructureConfig(infra *apisalicloud.InfrastructureConfig, node
 func ValidateInfrastructureConfigUpdate(oldConfig, newConfig *apisalicloud.InfrastructureConfig) field.ErrorList {
 	allErrs := field.ErrorList{}
 
+	var (
+		newZonesWithoutNatGatewayConfig = make([]apisalicloud.Zone, 0, len(newConfig.Networks.Zones))
+		oldZonesWithoutNatGatewayConfig = make([]apisalicloud.Zone, 0, len(oldConfig.Networks.Zones))
+
+		networksPath = field.NewPath("networks")
+	)
+
+	for i, zone := range newConfig.Networks.Zones {
+		newZonesWithoutNatGatewayConfig = append(newZonesWithoutNatGatewayConfig, apisalicloud.Zone{
+			Name:    zone.Name,
+			Worker:  zone.Worker,
+			Workers: zone.Workers,
+		})
+
+		allErrs = append(allErrs, ValidateNatGatewayConfig(zone.NatGateway, networksPath.Child("zones").Index(i).Child("natGateway"))...)
+	}
+
+	for i, zone := range oldConfig.Networks.Zones {
+		oldZonesWithoutNatGatewayConfig = append(oldZonesWithoutNatGatewayConfig, apisalicloud.Zone{
+			Name:    zone.Name,
+			Worker:  zone.Worker,
+			Workers: zone.Workers,
+		})
+
+		allErrs = append(allErrs, ValidateNatGatewayConfig(zone.NatGateway, networksPath.Child("zones").Index(i).Child("natGateway"))...)
```

Comment:
Do we need to validate old config?

Reply:
I don't think we need to validate old config, because after we update nat gateway config, we will use the new config to generate `TF` scripts, never to use the old scripts again. And `TF` will help us the reconcile the relationship between the new and old config.

Reply:
already removed validation of old config